lib/cypress-rails/init.rb

```diff
@@ -1,30 +1,44 @@
 module CypressRails
   class Init
-    DEFAULT_CONFIG = {
-      "screenshotsFolder" => "tmp/cypress_screenshots",
-      "videosFolder" => "tmp/cypress_videos",
-      "trashAssetsBeforeRuns" => false
-    }
+    DEFAULT_CONFIG = <<~CYPRESS_CONFIG
+    const { defineConfig } = require('cypress')
+
+    module.exports = defineConfig({
+      // setupNodeEvents can be defined in either
+      // the e2e or component configuration
+      e2e: {
+        setupNodeEvents(on, config) {
+          on('before:browser:launch', (browser = {}, launchOptions) => {
+            /* ... */
+          })
+        },
+      },
+      screenshotsFolder: "tmp/cypress_screenshots",
+      videosFolder: "tmp/cypress_videos",
+      trashAssetsBeforeRuns: false
+    })
+
+    CYPRESS_CONFIG
 
     def call(dir = Dir.pwd)
-      config_path = File.join(dir, "cypress.json")
-      json = JSON.pretty_generate(determine_new_config(config_path))
-      File.write(config_path, json)
+      config_path = File.join(dir, "cypress.config.js")
+      config_content = determine_new_config(config_path)
+      File.write(config_path, config_content)
       puts "Cypress config (re)initialized in #{config_path}"
     end
 
     private
 
     def determine_new_config(config_path)
       if File.exist?(config_path)
-        merge_existing_with_defaults(config_path)
+        merge_existing_with_defaults(config_path) 
       else
         DEFAULT_CONFIG
       end
     end
 
-    def merge_existing_with_defaults(json_path)
-      JSON.parse(File.read(json_path)).merge(DEFAULT_CONFIG).sort.to_h
+    def merge_existing_with_defaults(config_path)
+      File.read(config_path)
     end
```

Comment:
Since merging JS files isn't feasible for this gem to try to do here, I think we should avoid writing to the file ssytem entirely if `cypress.config.js` is present (as opposed to reading the file only to overwrite, as is done here) and instead `warn` that the file already exists and so we will not be re-initializing it.

Make sense?

Reply:
Thanks  for your comments!
Warn for users when running `cypress:init` and Cypress config already exist.

ref: https://github.com/testdouble/cypress-rails/pull/156/commits/c37692543fe9ae69815268820c73b7053747550b